src/git/orb.yml

```
@@ -69,11 +69,17 @@ commands:
 
             if [ -n "$CIRCLE_TAG" ]
             then
+              # The SHA1 for a given tag shouldn't change, so we *should* fail if this command doesn't work
               git reset --hard "$CIRCLE_SHA1"
               git checkout -q "$CIRCLE_TAG"
             elif [ -n "$CIRCLE_BRANCH" ]
             then
-              git reset --hard "$CIRCLE_SHA1"
+              # Don't fail the job if the hash isn't the latest
```

My initial read of the command felt misleading at first, because I read the `if` logic as "if the SHA1 didn't exist, then we will halt the job, so that means we'll fail then, so not what the comment says"

But after looking into it I now understand that `circleci-agent step halt` doesn't make the job fail, but instead make it end [gracefully](https://support.circleci.com/hc/en-us/articles/360015562253-Conditionally-end-a-running-job-gracefully) – which means it still ends up _green_, right?

So maybe we could find a clearer comment wording? 🤔  Something like this maybe:

```suggestion
              # Halt the job gracefully (early exit but green) if the hash isn't the latest
              # as the job will be later cancelled by another job building a more recent PRs anyway
```